### PR TITLE
feat: [PL-39458]: Added a tooltip to Delegate Instance Status column header

### PR DESCRIPTION
**Pull request by mayankVermaHarness**
### Summary

Jira: https://harness.atlassian.net/browse/PL-39458
[Usability]: Added a tooltip to Delegate Instance Status column header

#### Screenshots
image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

File: src/modules/27-platform/delegates/pages/delegates/DelegateListingItem.tsx [hunk 1]
```diff
@@ -336,7 +345,12 @@ export const DelegateListingItem: React.FC<DelegateProps> = props => {
         Cell: RenderVersion
       },
       {
-        Header: getString('platform.delegates.instanceStatus'),
+        Header: (
+          <>
+            {getString('platform.delegates.instanceStatus')}{' '}
```
Comment (namanharness) [inline src/modules/27-platform/delegates/pages/delegates/DelegateListingItem.tsx:350]: nit: `{' '}` intended?

**Review by namanharness — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

File: src/modules/27-platform/delegates/pages/delegates/DelegateListingItem.tsx (hunk 1 above)
Comment (litvinovstas) [reply src/modules/27-platform/delegates/pages/delegates/DelegateListingItem.tsx:350]: maybe make sense to put inside 'span', and provide some margin right ?

**Review by litvinovstas — APPROVED** [review]

File: src/modules/27-platform/delegates/pages/delegates/__tests__/DelegateListingItem.test.tsx [hunk 2]
```diff
@@ -31,12 +31,14 @@ jest.mock('services/cd-ng', () => ({
...
-    const { container } = render(
+    const { getByText } = render(
       <TestWrapper>
         <DelegatesListingItem data={delegateGroupsMock} />
       </TestWrapper>
     )
-    expect(container).toMatchSnapshot()
+
+    // Ensure that at least one of the delegateGroupIdentifier from mock data renders
+    expect(getByText(delegateGroupsMock[0].delegateGroupIdentifier)).toBeInTheDocument()
```
Comment (knagurski) [inline src/modules/27-platform/delegates/pages/delegates/__tests__/DelegateListingItem.test.tsx:41]: NICE! Thanks for killing off some snapshots!

File: src/modules/27-platform/delegates/pages/delegates/__tests__/DelegateListing.test.tsx [hunk 3]
```diff
@@ -317,6 +320,9 @@ describe('Test delegate buttons', () => {
       fireEvent.click(filterBtn!)
     })
 
-    expect(container).toMatchSnapshot()
+    // Ensure that "Filter" drawer opens
+    await waitFor(() => {
+      expect(screen.getByText('platform.delegates.delegateIdentifier')).toBeInTheDocument()
+    })
```
Comment (knagurski) [inline src/modules/27-platform/delegates/pages/delegates/__tests__/DelegateListing.test.tsx:326]: You can simplify `waitFor`/`getBy` combos by using `findBy` instead.
```suggestion
    expect(await screen.findByText('platform.delegates.delegateIdentifier')).toBeInTheDocument()
```

Does the exact same thing, so absolutely not a blocker.

File: src/modules/27-platform/delegates/pages/delegates/__tests__/DelegateListingItem.test.tsx [hunk 4]
```diff
@@ -31,12 +31,14 @@ jest.mock('services/cd-ng', () => ({
 const routesToDelegateDetails = jest.spyOn(routes, 'toDelegatesDetails')
 describe('Delegates Listing With Groups', () => {
   test('render data', () => {
-    const { container } = render(
+    const { getByText } = render(
```
Comment (knagurski) [inline src/modules/27-platform/delegates/pages/delegates/__tests__/DelegateListingItem.test.tsx:34]: Would it be worth getting this from `screen` instead?

**Review by knagurski — APPROVED** [review]
Good stuff. Just some suggestions, but nothing stopping a merge

File: src/modules/27-platform/delegates/pages/delegates/__tests__/DelegateListing.test.tsx (hunk 3 above)
Comment (mayankVermaHarness) [reply src/modules/27-platform/delegates/pages/delegates/__tests__/DelegateListing.test.tsx:326]: Thanks! I'll keep that in mind from next time onwards.

File: src/modules/27-platform/delegates/pages/delegates/__tests__/DelegateListingItem.test.tsx (hunk 4 above)
Comment (mayankVermaHarness) [reply src/modules/27-platform/delegates/pages/delegates/__tests__/DelegateListingItem.test.tsx:34]: I have seen that usually screen is useful when something renders in a portal (or OUTSIDE the `container` of the `render` in question.)

For all other use cases—while `screen` can still be used, but isn't necessary.

**Pull request merged**